keps/sig-release/20190118-creating-packages.md

```diff
@@ -0,0 +1,417 @@
+---
+kep-number: $i
+title: Packaging of kubernetes artefacts
+authors:
+  - "@hoegaarden"
+owning-sig: sig-release
+participating-sigs:
+  - sig-release
+  - sig-cluster-lifecycle
+reviewers:
+  - TBD
+approvers:
+  - TBD
+editor: TBD
+creation-date: 2019-01-18
+last-updated: 2019-01-18
+status: provisional
+see-also:
+  - ""
+replaces:
+  - ""
+superseded-by:
+  - "" 
+---
+
+# Packaging kubernetes artefacts
+
+## Table of Contents
+
+* [Packaging kubernetes artefacts](#packaging-kubernetes-artefacts)
+   * [Table of Contents](#table-of-contents)
+   * [Release Signoff Checklist](#release-signoff-checklist)
+   * [Summary](#summary)
+   * [Motivation](#motivation)
+      * [Goals](#goals)
+      * [Non-Goals](#non-goals)
+   * [Proposal](#proposal)
+      * [Risks and Mitigations](#risks-and-mitigations)
+   * [Design Details](#design-details)
+      * [Test Plan](#test-plan)
+      * [Graduation Criteria](#graduation-criteria)
+         * [From nil to k/k, Introduce the new mechanism into k/k](#from-nil-to-kk-introduce-the-new-mechanism-into-kk)
+         * [Deprecate all other options to build packages](#deprecate-all-other-options-to-build-packages)
+         * [Make this the only available way to build packages](#make-this-the-only-available-way-to-build-packages)
+      * [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+         * [Upgrade](#upgrade)
+      * [Version Skew Strategy](#version-skew-strategy)
+   * [Implementation History](#implementation-history)
+   * [Drawbacks [optional]](#drawbacks-optional)
+   * [Alternatives [optional]](#alternatives-optional)
+   * [Infrastructure Needed [optional]](#infrastructure-needed-optional)
+
+<!--
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+-->
+
+<!--
+**ACTION REQUIRED:** There must be an issue in
+[kubernetes/enhancements](https://github.com/kubernetes/enhancements/issues)
+referencing this KEP and targeting a release milestone **before [Enhancement
+Freeze](https://github.com/kubernetes/sig-release/tree/master/releases) of the
+targeted release**.
+-->
+
+## Release Signoff Checklist
+
+Check these off as they are completed for the Release Team to track. These checklist items _must_ be updated for the enhancement to be released.
+
+- [ ] k/enhancements issue in release milestone and linked to KEP (insert link here)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+
+## Summary
+
+There are currently different ways to build packages (`deb`s and
```

Comment:
i didn't realize there are multiple ways to build the debs/rpms. the only way i have been building them for the past several patches is using k/release/debian/jenkins.sh and k/release/rpm/docker-build.sh. since all other mechanisms haven't been in use, should we just deprecate all of them except one method?

Reply:
That's one goal, deprecate all except one.
However, I want to be careful with that and only do it when
- the new mechanism is working well
- all the places (in CI and whatnot) where e.g. the `bazel` mechanism or the packages generated by it have been identified